opensrp-app/src/main/java/org/smartregister/commonregistry/CommonRepository.java

```diff
@@ -60,10 +99,11 @@ public CommonRepository(String tablename, String[] columns) {
             if (i == 0) {
                 common_SQL = common_SQL + ", ";
             }
+
             if (i != columns.length - 1) {
-                common_SQL = common_SQL + columns[i] + " VARCHAR,";
+                common_SQL = common_SQL + columns[i].getName() + " " + columns[i].getDataType() + ",";
```

Comment:
It seems implementation of `defaultValue` and `length` is missing

Reply:
The Length is inside a for loop that is not shown here. The default value has been defined on the SQL, also not shown here

Reply:
Added